### DISCUSSION TITLE
Gratkit filament dryer

### DISCUSSION THREAD

**Discussion by rknobbe** [make-all/tuya-local#1911] [Q&A] [answered]
I've got Tuya Local working for a few other devices, and I'm trying to add my new Grafkit filament dryer. I see a file in the tree (custom_components/tuya_local/devices/gratkit_filament_dryer.yaml) that looks like _somebody_ else must be using it.  I can't get the add device wizard to pick it up.  Yes, I have a Tuya Developers account and yes I've followed the generic setup instructions.

**Comment by rknobbe** [comment]
it's entirely possible I've got the wrong IP address for the box because the error messages indicate "Device Unreachable". But I can't figure out how to get the device to tell me either its MAC or assigned IP addresses. I've tried with each of the Tuya devices that show up in my DHCP server's lease list, and none of them recognize the private key.

Here are the Homeassistant debug messages

This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:433
integration: Tuya Local (documentation, issues)
First occurred: May 12, 2024 at 10:01:16 AM (19 occurrences)
Last logged: May 12, 2024 at 7:12:53 PM

Test protocol error 914: Check device key or version
Test protocol error 905: Network Error: Device Unreachable
Test protocol error 901: Network Error: Unable to Connect
Test protocol error 904: Unexpected Payload from Device

**Reply by make-all** [reply]
The protocol version is also indicated by the first error message.

**Reply by rknobbe** [reply]
I had the checkbox set to "auto", for what that's worth.

**Reply by make-all** [reply, marked as answer]
auto does not always work (generally only for the older 3.3 and 3.1 devices). And even when it does work, it makes recovery after network outages take much longer and sometimes stall.

**Reply by rknobbe** [reply]
Thanks for the info. I figured out the IP address and got it to work with protocol version 3.5. All functions are exposed as entities now in HomeAssistant!
